Framework/Algorithms/test/CalculateAsymmetryTest.h

```diff
@@ -0,0 +1,333 @@
+#ifndef MUONCALCULATEASYMMETRYTEST_H_
+#define MUONCALCULATEASYMMETRYTEST_H_
+
+#include <cxxtest/TestSuite.h>
+#include "MantidAlgorithms/CalculateAsymmetry.h"
+#include "MantidKernel/PhysicalConstants.h"
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidHistogramData/LinearGenerator.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+
+using namespace Mantid::API;
+using Mantid::MantidVec;
+using Mantid::Algorithms::CalculateAsymmetry;
+
+const std::string outputName = "CalculateAsymmetry_Output";
+
+namespace {
+MatrixWorkspace_sptr createWorkspace(size_t nspec, size_t maxt) {
+
+  // Create a fake muon dataset
+  double a = 10.0; // Amplitude of the oscillations
+  double w = 5.0;  // Frequency of the oscillations
+  double tau = Mantid::PhysicalConstants::MuonLifetime *
+               1e6; // Muon life time in microseconds
+  double phi = 0.1;
+  MantidVec X;
+  MantidVec Y;
+  MantidVec E;
+  for (size_t s = 0; s < nspec; s++) {
+    for (size_t t = 0; t <= maxt; t++) {
+      double x = 10. * static_cast<double>(t) / static_cast<double>(maxt);
+      double e = exp(-x / tau);
+      X.push_back(x);
+      Y.push_back(20. * (1.0 + a * cos(w * x + phi)) * e);
+      E.push_back(0.005);
+    }
+  }
+
+  auto createWS = AlgorithmManager::Instance().create("CreateWorkspace");
+  createWS->initialize();
+  createWS->setChild(true);
+  createWS->setProperty("DataX", X);
+  createWS->setProperty("DataY", Y);
+  createWS->setProperty("DataE", E);
+  createWS->setProperty("NSpec", static_cast<int>(nspec));
+  createWS->setPropertyValue("OutputWorkspace", "ws");
+  createWS->execute();
+  MatrixWorkspace_sptr ws = createWS->getProperty("OutputWorkspace");
+  // Add  number of good frames
+  ws->mutableRun().addProperty("goodfrm", 10);
+  return ws;
+}
+}
+
+class CalculateAsymmetryTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static CalculateAsymmetryTest *createSuite() {
+    return new CalculateAsymmetryTest();
+  }
+  static void destroySuite(CalculateAsymmetryTest *suite) { delete suite; }
+
+  CalculateAsymmetryTest() { FrameworkManager::Instance(); }
+
+  void testInit() {
+    IAlgorithm_sptr alg =
+        AlgorithmManager::Instance().create("CalculateAsymmetry");
+    alg->initialize();
+    TS_ASSERT(alg->isInitialized())
+  }
+
+  void test_Execute() {
+
+    auto ws = createWorkspace(1, 50);
```

Comment:
Some of the tests are interested in multiple spectra. To save computation I have used as few spectra as possible each time. I think there are a couple that could be reduced to one spectrum.